reinforcement_learning/bindings/cs/rl.net/RankingResponse.cs

```diff
@@ -0,0 +1,131 @@
+
+        // TODO: Do we want to keep this consistent with .NET Collection idiom, or with the
+        // native RL Library naming (Size)?
+        public long Count
+        {
+            get
+            {
+                ulong unsignedSize = GetRankingSize(this.NativeHandle).ToUInt64();
+                Debug.Assert(unsignedSize < Int64.MaxValue, "We do not support collections with size larger than _I64_MAX/Int64.MaxValue");
+    
+                return (long)unsignedSize;
+            }
+        }
+
+        // TODO: Why does this method call, which seems like a "get" of a value, have an API status?
+        public bool TryGetChosenAction(out long action, ApiStatus status = null)
+        {
+            action = -1;
+            UIntPtr chosenAction;
+            int result = GetRankingChosenAction(this.NativeHandle, out chosenAction, status.ToNativeHandleOrNullptr());
+
+            if (result == NativeMethods.SuccessStatus)
+            {
+                return false;
+            }
+
+            action = (long)(chosenAction.ToUInt64());
+            return true;
+        }
+
+        private class RankingResponseEnumerator : NativeObject<RankingResponseEnumerator>, IEnumerator<ActionProbability>
+        {
+            [DllImport("rl.net.native.dll")]
+            private static extern IntPtr CreateRankingEnumeratorAdapter(IntPtr rankingResponse);
+
+            private static New<RankingResponseEnumerator> BindConstructorArguments(RankingResponse rankingResponse)
+            {
+                return new New<RankingResponseEnumerator>(() => CreateRankingEnumeratorAdapter(rankingResponse.NativeHandle));
+            }
+
+            [DllImport("rl.net.native.dll")]
+            private static extern void DeleteRankingEnumeratorAdapter(IntPtr rankingEnumeratorAdapter);
+
+            [DllImport("rl.net.native.dll")]
+            private static extern int RankingEnumeratorMoveNext(IntPtr rankingEnumeratorAdapter);
+        
+            [DllImport("rl.net.native.dll")]
+            private static extern ActionProbability GetRankingEnumeratorCurrent(IntPtr rankingEnumeratorAdapter);
+
+            public RankingResponseEnumerator(RankingResponse rankingResponse) : base(BindConstructorArguments(rankingResponse), new Delete<RankingResponseEnumerator>(DeleteRankingEnumeratorAdapter))
+            {
+            }
+
+            public ActionProbability Current
+            {
+                get
+                {
+                    return GetRankingEnumeratorCurrent(this.NativeHandle);
+                }
+            }
+
+            object System.Collections.IEnumerator.Current => this.Current;
+
+            public bool MoveNext()
+            {
+                // The contract of result is to return 1 if true, 0 if false.
+                int result = RankingEnumeratorMoveNext(this.NativeHandle);
+
+                return result == 1;
+            }
+
+            public void Reset()
```

Comment:
`Reset` is part of the `IEnumerable` interface. I cannot avoid having the method, even though it is not a possible operation, but the usual way of specifying that Reset is not supported is to throw `NotSupportedException`. `foreach` does not use Reset - it's a one-shot pass through the collection, so I have never seen this construction having an issue.

Reply:
Ok, I guess it would be an issue if someone explicitly call the `Reset` method, what is not recommended. (Just for sanity, `Reset` is part of `IEnumerator`, not `IEnumerable`)